### PR TITLE
details about implementing Tensor in paddle

### PR DESCRIPTION
**Pull request by QiJune**
This documentation will be edit again after tomorrow's discussion. 

Related with

1.  #2129 
1.  #2130 

[This](https://github.com/QiJune/Paddle/blob/51717d8fe4f88f2ecad8045bd33775d7fe54acf7/paddle/majel/README.md) could be more readable.

### CODE REVIEW

File: paddle/majel/README.md [hunk 1]
```diff
@@ -95,32 +95,170 @@ Because `variant` may be thought of as "multi-type, single value", we can utiliz
 
 ## implement Tensor in Paddle
 
+We aim to implement an independent tensor library, which provides flexiable and efficient tensor operations and can run on muti-devices(CPU and GPU). In brief, there are mainly two aspects to consider. First, we have to make encapsulation of the hardware for the tensor library, including device memory, computing handles, cuda streams and so on. Second, after the tensor libaray is able to access the hardware resources, we should make carefully design and abstraction of the upper levels and provide flexiable interfaces.
```
Comment (gangliao) [inline paddle/majel/README.md]: flexiable  -> flexible

Comment (gangliao) [inline paddle/majel/README.md]: to consider -> to be considered

Comment (wangkuiyi) [inline paddle/majel/README.md]: What is a "computing handle"?

Comment (wangkuiyi) [inline paddle/majel/README.md]: cuda => CUDA

File: paddle/majel/README.md [hunk 2]
```diff
@@ -95,32 +95,170 @@ Because `variant` may be thought of as "multi-type, single value", we can utiliz
 
 ## implement Tensor in Paddle
 
+We aim to implement an independent tensor library, which provides flexiable and efficient tensor operations and can run on muti-devices(CPU and GPU). In brief, there are mainly two aspects to consider. First, we have to make encapsulation of the hardware for the tensor library, including device memory, computing handles, cuda streams and so on. Second, after the tensor libaray is able to access the hardware resources, we should make carefully design and abstraction of the upper levels and provide flexiable interfaces.  
+
 Before writing code, please make sure you already look through Majel Source Code and grabbed the design philosophy of `DArray` in Majel.
 
-To assign subtasks to our colleagues, we have to discuss how to divide it to independent subtasks.
+### Resources Manegement
```
Comment (gangliao) [inline paddle/majel/README.md]: Manegement  -> Management

File: paddle/majel/README.md (hunk 1 above)
Comment (wangkuiyi) [inline paddle/majel/README.md]: The paragraph is not dense enough.  For example, "In brief, there are mainly two aspects to consider. " doesn't seem necessary.  An example change is as follows:

> We want to create a Tensor class to replace Vector and Matrix, and to support high-dimensional data. We want to accelerate operations on Tensor using CUDA, so we need CPU and GPU memory allocation and CUDA kernel launching. We also want to make sure that the Tensor interface is friendly to its callers.

Comment (QiJune) [inline paddle/majel/README.md]: The computing handle  is a pointer  to an opaque structure holding the some CUDA library context, such as cublasHandle, cudnnHandle, cufftHandle

File: paddle/majel/README.md (hunk 2 above)
Comment (QiJune) [inline paddle/majel/README.md]: Done.

File: paddle/majel/README.md (hunk 1 above)
Comment (QiJune) [inline paddle/majel/README.md]: Done.

Comment (QiJune) [inline paddle/majel/README.md]: Done.

Comment (QiJune) [inline paddle/majel/README.md]: Done.

**Pull request closed without merging**